config/k8s-patroni.txt

```diff
@@ -9,57 +9,58 @@ These configmaps are automatically discovered by the zabbix-agent.
-      PATRONI_ARCHIVE_COMMAND_SET="$(jq -r '.postgresql.parameters.archive_command != null' /tmp/patroni-config.json)"
-
-      WALG_LAST_BACKUP="$(envdir /etc/wal-g.d/env wal-g backup-list --json 2> /dev/null | jq '.[-1] // {}' -c)"
-      WALG_TIMELINE="$(envdir /etc/wal-g.d/env wal-g wal-verify timeline --json 2>/dev/null)"
-      WALG_INTEGRITY="$(envdir /etc/wal-g.d/env wal-g wal-verify integrity --json 2>/dev/null)"
-
-      REQUEST_JSON="$(echo "{\"cluster_status\": ${CLUSTER_STATUS}, \"patroni_archive_enabled\": ${PATRONI_ARCHIVE_ENABLED}, \"patroni_archive_command_set\": ${PATRONI_ARCHIVE_COMMAND_SET}, \"walg_last_backup\": ${WALG_LAST_BACKUP}, \"walg_timeline\": ${WALG_TIMELINE}, \"walg_integrity\": ${WALG_INTEGRITY}}" | jq -c .)"
-
-      PAYLOAD="[{\"op\" : \"replace\", \"path\" : \"/data/status\", \"value\" : \"${REQUEST_JSON//\"/\\\"}\" }]"
-      curl --cacert ${CA_CERT} --header @/tmp/auth --header "Content-Type: application/json-patch+json" --request PATCH --data "${PAYLOAD}" \
-      https://kubernetes.default.svc:443/api/v1/namespaces/${POD_NAMESPACE}/configmaps/status-${STATUS_NAME}
-
-      echo "status updated"
-    else
-      echo "Not eligable, I am replica"
-    fi
+#!/bin/bash
+umask 0077
+ROLE=$(curl -s http://localhost:8008 | jq -r .role)
+# there is only one master, use this to prevent double status checks.
+if test $ROLE == "master"; then
+  CA_CERT="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt";
+  sed -e 's/^/Authorization: Bearer /' /var/run/secrets/kubernetes.io/serviceaccount/token >/tmp/auth
+  CLUSTER_STATUS=$(curl localhost:8008/cluster)
+  curl localhost:8008/config > /tmp/patroni-config.json
+  PATRONI_ARCHIVE_ENABLED="$(jq -r '.postgresql.parameters.archive_mode == "on"' /tmp/patroni-config.json)"
+  PATRONI_ARCHIVE_COMMAND_SET="$(jq -r '.postgresql.parameters.archive_command != null' /tmp/patroni-config.json)"
+
+  WALG_LAST_BACKUP="$(envdir /etc/wal-g.d/env wal-g backup-list --json 2> /dev/null | jq '.[-1] // {}' -c)"
+  WALG_TIMELINE="$(envdir /etc/wal-g.d/env wal-g wal-verify timeline --json 2>/dev/null)"
+  WALG_INTEGRITY="$(envdir /etc/wal-g.d/env wal-g wal-verify integrity --json 2>/dev/null)"
+
+  REQUEST_JSON="$(echo "{\"cluster_status\": ${CLUSTER_STATUS}, \"patroni_archive_enabled\": ${PATRONI_ARCHIVE_ENABLED}, \"patroni_archive_command_set\": ${PATRONI_ARCHIVE_COMMAND_SET}, \"walg_last_backup\": ${WALG_LAST_BACKUP}, \"walg_timeline\": ${WALG_TIMELINE}, \"walg_integrity\": ${WALG_INTEGRITY}}" | jq -c .)"
+
+  PAYLOAD="[{\"op\" : \"replace\", \"path\" : \"/data/status\", \"value\" : \"${REQUEST_JSON//\"/\\\"}\" }]"
+  curl --cacert ${CA_CERT} --header @/tmp/auth --header "Content-Type: application/json-patch+json" --request PATCH --data "${PAYLOAD}" \
+  https://kubernetes.default.svc:443/api/v1/namespaces/${POD_NAMESPACE}/configmaps/status-${STATUS_NAME}
+
+  echo "status updated"
+else
+  echo "Not eligable, I am replica"
+fi
```

Comment:
Nice.

Can you indent this by 4 spaces?

Reply:
resolved in latest commit